### ISSUE TITLE
Issues selecting data in handlebars to pass through PATCH

### ISSUE THREAD

**Issue by rowlandhill** [ga-wdi-boston/full-stack-project#859]
I've looked through other queue requests, but none of the solutions seem to work for what I'm doing.  I'm able to display every single recipe created by the user.  I have an update field below each recipe that has input forms for the user to update.  

I'm trying to allow the user to update the recipes when they see them on the page (after handlebars renders the html).  After I fill in the update forms, I'm failing to then select the particular field I just updated using jQuery, and pass that through the `updateRecipe` function.  Whatever code I write seems to just grab every single recipe that is on the page (I've been living in the chrome console, trying to select the individual recipe, but whatever code I try, it selects every data-id, not the one I want).  Here's my code:

handlebars:

```js
{{#each recipes as |recipe|}}

  <div data-id="{{recipe.id}}">
    <ul>
      <li><h4>How to make {{recipe.name}}</h4></li>
      <li class="desc">Description: {{recipe.description}}</li>
      <li class="desc">Servings: {{recipe.servings}}</li>
      <li class="desc">Recipe ID: {{recipe.id}}</li>
  </ul>
  <form class="update" id="{{recipe.id}}" data-id="{{recipe.id}}">

    <div class="container-fluid update-fields">
      <br>
       <input class="form-control" type="text" name="recipe[name]" placeholder="recipe name">
       <div class="form-group">
      <p>update description</p>
     <textarea class="form-control" name="recipe[description]" rows="7"></textarea>
    </div>
    <div class="form-group">
      <input class="form-control" name="recipe[servings]" type="text" value="" placeholder="update serving size">
    </div>
    <input type="submit" class="btn btn-default" name="submit" value="update">
    </form>
   </div>
 </div>
```

Here's the code I'm trying to run after handlebars is executed.  `getAllRecipes` is the function that calls handlebars and displays them all.

`  $('.update').on('submit', events.onUpdateRecipe)` is the code I'm running currently to try and grab the updated recipe, but the page just refreshes.  I'm guessing the button inside the handlebars HTML is not connected/linked to this code at all, but I'm not sure why:

```js
const getAllRecipesSuccess = (data) => {
  console.log(data)
  const showRecipesHtml = showRecipesTemplate({ recipes: data.recipes })
  $('.content').append(showRecipesHtml)
  $('.update').on('submit', events.onUpdateRecipe)
}
```

That should take us to events.js to run this:

```js
const onUpdateRecipe = (event) => {
  event.preventDefault()
  const data = getFormFields(event.target)
  const newRecipe = $(event.target).attr('data-id')
  api.updateRecipe(data, newRecipe)
    .then(ui.updateRecipeSuccess)
    .catch(ui.updateRecipeFailure)
}
```

And then that should lead us to api.js for this:

```js
const updateRecipe = (data, newRecipe) => {
  // console.log(data + 'PATCH TEST DATA')
  return $.ajax({
    url: config.apiOrigin + '/recipes/' + newRecipe,
    method: 'PATCH',
    headers: {
      Authorization: 'Token token=' + store.user.token
    },
    data
  })
}
```

I think the issue here is that the function isn't firing, but I don't know why and I'm not sure where else to look.

**Comment by jordanallain** [comment]
have you tried using `this` to limit the selection to the current one?

**Comment by rowlandhill** [comment]
I didn't try `this` (that probably would have helped me narrow it down), but I ended up fixing the glitch by moving `onUpdateRecipe` from the events.js file into the ui.js file.  For some reason, it was not firing when it had to go back to the events.js file.  Once I moved it, the update fired, but then I had to go in and fix some authenticate issues on the backend that are not related to this request (had to change where the RecipeController was inheriting from)

All good on this issue.  Thanks, Jordan!  I'll close this out.
